lib/grape/api_instance.rb

```diff
@@ -0,0 +1,233 @@
+require 'grape/router'
+
+module Grape
+  # The API Instance class, is the engine behind Grape::API. Each class that inherits
+  # from this will represent a different API
+  class APIInstance
```

Comment:
Would this be better called `Grape::API::Instance`? Or `Grape::SomethingSomething::API` (to match how we have `Grape::DSL::API`)? Just thinking out loud, no strong feelings.

The naming could be important because if someone is having trouble with the new `Grape::API` they could just inherit from `Grape::API::Instance`, making the diff prettier ;)

Reply:
I'm happy with that naming. Regarding the solution if someone's having trouble: Should I add an entry that says something among those lines in `updating.md` ?